lib/features/server_settings/data/network/server_settings_api.dart

```diff
@@ -44,11 +47,22 @@ class ServerSettingsAPI with HandleSetErrorMixin {
   }
 
   Future<TMailServerSettings> updateServerSettings(
+    Session session,
     AccountId accountId, 
     TMailServerSettings serverSettings
   ) async {
     final processingInvocation = ProcessingInvocation();
 
+    if (session.isLanguageReadOnly(accountId)) {
+      serverSettings = TMailServerSettings(
+        id: serverSettings.id,
+        settings: TMailServerSettingOptions(
+          alwaysReadReceipts: serverSettings.settings?.isAlwaysReadReceipts,
+          displaySenderPriority: serverSettings.settings?.isDisplaySenderPriority,
+        ),
+      );
+    }
```

Comment:
We should write extensions for easy maintenance later.

```suggestion
extension TMailServerSettingsExtension on TMailServerSettings {

  TMailServerSettings normalized(Session session, AccountId accountId) {
    if (session.isLanguageReadOnly(accountId)) {
      return TMailServerSettings(
        id: id,
        settings: TMailServerSettingOptions(
          alwaysReadReceipts: settings?.alwaysReadReceipts,
          displaySenderPriority: settings?.displaySenderPriority,
        ),
      );
    }
    return this;
  }
}
```

Reply:
Done